File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/DNDChoice.java
package com.thewizrd.shared_resources.helpers;

import android.util.SparseArray;

public enum DNDChoice {
    OFF(0),
    PRIORITY(1),
    ALARMS(2),
    SILENCE(3);

    private final int value;

    public int getValue() {
        return value;
    }

    private DNDChoice(int value) {
        this.value = value;
    }

    private static SparseArray<DNDChoice> map = new SparseArray<>();

    static {
        for (DNDChoice choice : values()) {
            map.put(choice.value, choice);
        }
    }

    public static DNDChoice valueOf(int value) {
        return map.get(value);
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/actions/DNDChoice.kt
package com.thewizrd.shared_resources.actions

import android.util.SparseArray

enum class DNDChoice(val value: Int) {
    OFF(0),
    PRIORITY(1),
    ALARMS(2),
    SILENCE(3);

    companion object {
        private val map = SparseArray<DNDChoice>()

        fun valueOf(value: Int): DNDChoice {
            return map[value]
        }

        init {
            for (choice in values()) {
                map.put(choice.value, choice)
            }
        }
    }
}